frontend/src/components/employees/index.tsx

```diff
@@ -1,11 +1,157 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import toastr from 'toastr';
+
+import Button from '@material-ui/core/Button';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableBody from '@material-ui/core/TableBody';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+
+import Clear from '@material-ui/icons/Clear';
+import EditOutlined from '@material-ui/icons/EditOutlined';
+
+import { fetchStaff, selectStaff } from '../../features/staffSlice';
+import 'toastr/build/toastr.min.css';
+import {
+  STAFF_FETCH_URL,
+  StaffData,
+  TableColumn,
+} from '../../constants/employees';
+import { FULLTIME_HOURS } from '../../constants/global';
 import employeesStyle from './employees.module.scss';
 
-const Employees = () => (
-  <div className={employeesStyle.employees}>
-    <h1> Employees </h1>
-    <p> Employees component </p>
+// A placeholder for authentication token
+const getAuthenticationToken = (): string => 'xx';
+
+/**
+ * Functions which are called when add/edit/delete buttons are pressed.
+ * Note that the signature of addUser is different from editUser/deleteUser.
+ */
+
+const addUser = () => toastr.info('Add user');
+
+const editUser = (id: number) => () => toastr.info(`User ${id} edited`);
+
+const deleteUser = (id: number) => () => toastr.info(`User ${id} deleted`);
+
+/**
+ * Functions which extract the data to be displayed in the table columns
+ * from StaffData.
+ */
+
+const getName = ({ firstName, surname }: StaffData) =>
+  `${firstName} ${surname}`;
+
+const getJobTitle = ({ jobTitle }: StaffData) => jobTitle;
+
+const partFullTime = (fullTimeHours: number) => ({
+  contractedHours,
+}: StaffData) => (contractedHours >= fullTimeHours ? 'full' : 'part');
+
+const editUserButton = (editFunction: (o: number) => () => any) => ({
```

Comment:
I'm pretty new to TS and new JavaScript functions... can you explain this line to me please? 😊

Reply:
It defined a new javascript function which takes a single argument, which is a function 
that takes a number as an argument and returns a function with no arguments and has any return type. Sounds confusing, I know, but it's simple once you're used to it.

If you have a function
```
const f = x1 => x2 => [etc.] => xn => result 
```
it is a 'curried' function where you can think of x1,...,xn as arguments and the part after the last `=>` as the result.
You can provide all arguments immediately `f(x1)(x2)[etc.](xn)` and get the result, or you can apply it partially `f(x1)(x2)[etc.](x(n-k))` and get a curried function from `x(n-k+1)=>...=>xn=>result`